internal/controller/pkg/revision/dependency.go

```diff
@@ -137,6 +137,24 @@ func (m *PackageDependencyManager) Resolve(ctx context.Context, pkg runtime.Obje
 		Dependencies: sources,
 	}
 
+	// Delete packages in lock with same name and distinct source
+	// This is a corner case when source is updated but image SHA is not (i.e. relocate same image
+	// to another registry)
+	for _, lp := range lock.Packages {
+		if self.Name == lp.Name &&
+			self.Type == lp.Type &&
+			self.Source != lp.Identifier() {
```

Comment:
Nit: I wouldn't worry about wrapping this. Doesn't seem like the unwrapped line would be very long compared to the rest of the file.

Reply:
Done